bin/arg.ml

```diff
@@ -25,3 +25,94 @@ let path = Path.conv
 
 [@@@ocaml.warning "-32"]
 let file = path
+
+module Dep : sig
+  type t = Dune_file.Dep_conf.t
+
+  val conv : t conv
+
+  val file : string -> t
+  val alias : string -> t
+  val alias_rec : string -> t
+
+  val to_string_maybe_quoted : t -> string
+end = struct
+  module Dep_conf = Dune_file.Dep_conf
+
+  type t = Dep_conf.t
+
+  let file s =
+    Dep_conf.File (String_with_vars.make_text Loc.none s)
+
+  let alias s =
+    Dep_conf.Alias (String_with_vars.make_text Loc.none s)
+
+  let alias_rec s =
+    Dep_conf.Alias_rec (String_with_vars.make_text Loc.none s)
+
+  let parse_alias s =
+    if not (String.is_prefix s ~prefix:"@") then
+      None
+    else
+      let pos, recursive =
+        if String.length s >= 2 && s.[1] = '@' then
+          (2, false)
+        else
+          (1, true)
+      in
+      let s = String.drop s pos in
+      Some (recursive, s)
+
+  let dep_parser =
+    Syntax.set Stanza.syntax Stanza.latest_version
+      Dep_conf.decode
+
+  let parser s =
+    match parse_alias s with
+    | Some (true, s) ->
+      `Ok (alias_rec s)
+    | Some (false, s) ->
+      `Ok (alias s)
+    | None ->
+      match
+        Dune_lang.Decoder.parse dep_parser Univ_map.empty
+          (Dune_lang.parse_string ~fname:"command line"
+             ~mode:Dune_lang.Parser.Mode.Single s)
+      with
+      | x -> `Ok x
+      | exception (User_error.E msg) ->
+        `Error (User_message.to_string msg)
```

Comment:
As mentioned previously, I'm not entirely sure this prints the errors correctly.

Reply:
I added a test showing the kind of errors that come out.

Reply:
Yeah, it doesn't look all that good but I suppose it's good enough for now.